src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -85,21 +94,116 @@ public function setCallback($callback = null)
     }
 
     /**
+     * @param string $content
+     *
+     * @return Response
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function setContent($content)
+    {
+        if (!is_string($content)) {
+            throw new \InvalidArgumentException('Content is not a string. Use addData for adding data');
+        }
+
+        $data = json_decode($content);
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            throw new \InvalidArgumentException(json_last_error_msg());
```

Comment:
@xabbuh Thanks for the poyfill remark. Looks like a very handy library. I think all the php-polyfills classes are bootstrapped with symfony 2.8 right?